pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -291,41 +293,24 @@ func (s *ServiceEntryStore) serviceEntryHandler(old, curr model.Config, event mo
 		return
 	}
 
-	if fullPush {
-		// When doing a full push, for added and updated services trigger an eds update
-		// so that endpoint shards are updated.
-		var instances []*model.ServiceInstance
-		if len(addedSvcs) > 0 {
-			instances = append(instances, convertInstances(curr, addedSvcs)...)
-		}
-		if len(updatedSvcs) > 0 {
-			instances = append(instances, convertInstances(curr, updatedSvcs)...)
-		}
-		if len(unchangedSvcs) > 0 {
-			currentServiceEntry := curr.Spec.(*networking.ServiceEntry)
-			oldServiceEntry := old.Spec.(*networking.ServiceEntry)
-			// Non DNS service entries are sent via EDS. So we should compare and update if such endpoints change.
-			if currentServiceEntry.Resolution != networking.ServiceEntry_DNS {
-				if !reflect.DeepEqual(currentServiceEntry.Endpoints, oldServiceEntry.Endpoints) {
-					instances = append(instances, convertInstances(curr, unchangedSvcs)...)
-				}
-			}
-		}
-		s.edsUpdate(instances, true)
-
-		// If service entry is deleted, cleanup endpoint shards for services.
-		for _, svc := range deletedSvcs {
-			s.XdsUpdater.SvcUpdate(s.Cluster(), string(svc.Hostname), svc.Attributes.Namespace, model.EventDelete)
-		}
+	// When doing a full push, the added, updated, unchanged services trigger an eds update
+	// so that endpoint shards are updated.
+	allServices := make([]*model.Service, 0, len(addedSvcs)+len(updatedSvcs)+len(unchangedSvcs))
+	allServices = append(allServices, addedSvcs...)
+	allServices = append(allServices, updatedSvcs...)
+	allServices = append(allServices, unchangedSvcs...)
+	instances := convertServiceEntryToInstances(curr, addedSvcs)
+	s.edsUpdate(instances, false)
```

Comment:
Note: we need all the instances here

Reply:
This is is same as existing implementation except that now we just update eds cache rather than doing a push?

Reply:
A little different for unchanged services

https://github.com/istio/istio/pull/26405/files#diff-9d8ccc56790e36dcbe7c7cd098f8d7d3L304-L313

It should be all service instances here as `edsUpdate` calls `DiscoveryServer.EDSUpdate` to update all the caches associated with a service

Reply:
Ok. But don't we have to check the resolution and skip DNS service entries??

Reply:
https://github.com/istio/istio/pull/26405#discussion_r469142655